### PR TITLE
refactor: Make invalid schema reporting optional

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

This allows us to validate and leverage type guarding even if we don't need any reporting

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

File: lib/util/schema.spec.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { z } from 'zod';
 import { logger } from '../../test/util';
 import * as memCache from './cache/memory';
-import { checkSchema, reportErrors } from './schema';
+import * as schema from './schema';
```
Comment (viceice) [inline lib/util/schema.spec.ts:4]: @zharinov can we prefer non star imports? i don't like those

Comment (zharinov) [reply lib/util/schema.spec.ts:4]: I think it's okay for this particular case: `schema.match(...)` looks most clear comparing to other alternatives to me. What alternative would you like?

Comment (viceice) [reply lib/util/schema.spec.ts:4]: `matchesSchema(...)` 😉

Comment (zharinov) [reply lib/util/schema.spec.ts:4]: I would like to see something shorter, though 🤔

Comment (zharinov) [reply lib/util/schema.spec.ts:4]: Seems like we'll use it a lot in future, so we need to think about naming

Comment (viceice) [reply lib/util/schema.spec.ts:4]: yes, 🤔

**Pull request merged**